src/package/midlrt/Microsoft.Windows.MidlRT.IdlGen.targets

```diff
@@ -0,0 +1,115 @@
+    We need a special target here, because the other Get*References  depends on ResolveReferences
+    being run first. However, MIDL runs before ResolveReferences, so this target uses the References
+    item, not the ReferencePath item.
+  -->
+  <Target Name="GetMidlRTDirectWinMDReferencesForIdl"
+          BeforeTargets="Midl"
+          Returns="@(MidlRTDirectWinMDReferencesForIdl);@(MidlRTDirectWinMDReferenceDirs)">
+    <ItemGroup>
+      <_MidlRTDirectWinMDReferencesForIdl Remove="@(_MidlRTDirectWinMDReferencesForIdl)" />
+      <_MidlRTDirectWinMDReferencesForIdl Include="@(Reference)" Condition="'%(Reference.Extension)' == '.winmd' and '%(Reference.IsSystemReference)' != 'true'" />
+      <MidlRTDirectWinMDReferencesForIdl Remove="@(MidlRTDirectWinMDReferencesForIdl)"/>
+      <MidlRTDirectWinMDReferencesForIdl Include="@(_MidlRTDirectWinMDReferencesForIdl)">
+        <WinMDPath>%(FullPath)</WinMDPath>
+      </MidlRTDirectWinMDReferencesForIdl>
+
+      <!-- Generate list of reference directories to pass into winmdidl.exe -->
+      <_MidlRTDirectWinMDReferenceDirs Include="@(MidlRTDirectWinMDReferencesForIdl->'%(RootDir)%(Directory)')"/>
+    </ItemGroup>
+
+    <!-- Remove any duplicates of file paths (caused by a nuget containing mulitple .winmd files) -->
+    <RemoveDuplicates Inputs="@(_MidlRTDirectWinMDReferenceDirs)">
+      <Output TaskParameter="Filtered" ItemName="MidlRTDirectWinMDReferenceDirs"/>
+    </RemoveDuplicates>
+    <Message Text="MidlRTDirectWinMDReferencesForIdl: @(MidlRTDirectWinMDReferencesForIdl->'%(WinMDPath)')" Importance="$(MidlRTVerbosity)"/>
+  </Target>
+
+  <Target Name="_GatherExpectedGeneratedFiles"
+          Condition="'@(MidlRTDirectWinMDReferencesForIdl)' != ''"
+          BeforeTargets="MidlRTMakeReferenceIdl"
+          Returns="@(_ExpectedGeneratedIdl)">
+    <ReadLinesFromFile File="$(_MidlRTIdlGenTrackerFile)" >
+      <Output TaskParameter="Lines" ItemName="_ExpectedGeneratedIdl"/>
+    </ReadLinesFromFile>
+
+    <!-- If no files we're read, then we haven't run this target before, so include a fake so the MidlRTMakeReferenceIdl target will execute -->
+     <ItemGroup Condition="'@(_ExpectedGeneratedIdl)'==''">
+      <_ExpectedGeneratedIdl Include="fake.idl"/>
+    </ItemGroup>
+  </Target>
+
+  <!--Build reference IDL from WinMD project references and dynamic library project references
+      Note that Condition is evaluated before DependsOnTargets are run -->
+  <Target Name="MidlRTMakeReferenceIdl"
+          Condition="'@(MidlRTDirectWinMDReferencesForIdl)' != ''"
+          AfterTargets="GetMidlRTDirectWinMDReferencesForIdl"
+          Inputs="@(MidlRTDirectWinMDReferencesForIdl)"
+          Outputs="@(_ExpectedGeneratedIdl)">
+    <PropertyGroup>
+      <WindowsSdkToolLocation Condition="'$(WindowsSdkToolLocation)'==''">$(WDKBinRoot)\$(PreferredToolArchitecture)</WindowsSdkToolLocation>
```

Comment:
why is this using a WDK property?

Reply:
i'll have to take a look, i think this is what's set in uap.props that points to where winmdidl is

Reply:
it was, but it looks like it was removed in the latest sdk, so i'll change this